orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -112,3 +115,38 @@ def _create_iobinding(io_binding, inputs, model, device):
 
     for value_info in model.graph.output:
         io_binding.bind_output(value_info.name, device.type, device_id=get_device_index(device))
+
+def check_for_name_collisions(ortmodule: torch.nn.Module, user_module: torch.nn.Module,
+                              ortmodule_non_method_attribute_names: List[str]):
+    """Raises if there are any common attributes between the user's model and ORTModule.
+
+    Args:
+        ortmodule: the ORTModule instance
+        user_module: the user's torch.nn.Module
+        ortmodule_non_method_attribute_names: any non method attribute names that ORTModule sets on itself
+
+    Raises:
+        UserWarning: If there are any overlapping methods between the ortmodule and user_module (except forward)
+    """
+
+    ortmodule_methods = dict(inspect.getmembers(ortmodule, predicate=inspect.ismethod))
+    torch_module_methods = dict(inspect.getmembers(torch.nn.Module, predicate=inspect.isfunction))
```

Comment:
yes, it is intentional here because we only want to check for user define methods that collide with ORTModule in this part of the function.
We check if the user has defined any non method attribute that collides with ORTModule using ```ortmodule_non_method_attribute_names``` in another part.

Secondly, ```torch.nn.Module``` is a class and not a class instance. So, it would not have any non method attributes (unless they were static), so it would be the same querying with and without ```inspect.isfunction```.